ffi/diplomat/src/displaynames.rs

```diff
@@ -0,0 +1,155 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[diplomat::bridge]
+pub mod ffi {
+    use alloc::boxed::Box;
+    use icu_displaynames::{LanguageDisplayNames, RegionDisplayNames};
+    use icu_displaynames::options::DisplayNamesOptions;
+    use diplomat_runtime::DiplomatWriteable;
+    use crate::errors::ffi::ICU4XError;
+    use crate::locale::ffi::ICU4XLocale;
+    use crate::provider::ffi::ICU4XDataProvider;
+    use writeable::Writeable;
+
+    //  FFI version of `LanguageDisplayNames`.
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu::displaynames::LanguageDisplayNames, Struct)]
+    pub struct ICU4XLanguageDisplayNames(pub LanguageDisplayNames);
+
+    //  FFI version of `RegionDisplayNames`.
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu::displaynames::RegionDisplayNames, Struct)]
+    pub struct ICU4XRegionDisplayNames(pub RegionDisplayNames);
+
+    //  FFI version of `DisplayNamesOptions`.
+    #[diplomat::rust_link(icu::displaynames::options::DisplayNamesOptions, Struct)]
+    pub struct ICU4XDisplayNamesOptions {
+        /// The optional formatting style to use for display name.
+        pub style: ICU4XStyle,
+        /// The fallback return when the system does not have the
+        /// requested display name, defaults to "code".
+        pub fallback: ICU4XFallback,
+        /// The language display kind, defaults to "dialect".
+        pub language_display: ICU4XLanguageDisplay,
+    }
+
+    // FFI version of `Style`.
+    #[diplomat::rust_link(icu::displaynames::options::Style, Enum)]
+    pub enum ICU4XStyle {
+        Auto,
+        Narrow,
+        Short,
+        Long,
+        Menu,
+    }
+
+    // FFI version of `Fallback`.
+    #[diplomat::rust_link(icu::displaynames::options::Fallback, Enum)]
```

Comment:
nit: you should be able to use `#[diplomat::enum_convert]` here, then you don't need the `From` impl

(you still need it for `ICU4XStyle` unfortunately because of the `Auto`)

Reply:
Done.